### PR TITLE
feat: add category listing page with SEO metadata and cache invalidation

### PR DESCRIPTION
**Pull request by gajda-w**
I want to merge this change because it adds category listing pages to the storefront, bringing categories to parity with the existing collection pages (SEO metadata, OG image, loading state, and cache-invalidating webhook). This closes a gap where categories had no dedicated browsable page, and reuses the shared listing-header/listing-description components extracted from collections so both features stay consistent and avoid duplicated UI code.

# Pull Request Checklist

- [ ] Test the changes locally to ensure they work as expected.
- [ ] Document the testing process and results in the pull request description. (Screen recording, screenshot etc)
- [ ] Include new tests for any new functionality or significant changes.
- [ ] Ensure that tests cover edge cases and potential failure points.
- [ ] Document the impact of the changes on the system, including potential risks and benefits.
- [ ] Provide a rollback plan in case the changes introduce critical issues.
- [ ] Update documentation to reflect any changes in functionality.

### CODE REVIEW

File: apps/marketplace/src/graphql/generated/client.ts [hunk 1]
```diff
@@ -4882,6 +4847,7 @@ export type CheckoutErrorCode =
   | 'NOT_FOUND'
   | 'NO_LINES'
   | 'PAYMENT_ERROR'
+  | 'PRICE_OVERRIDE_REASON_WITHOUT_OVERRIDE'
```
Comment (lukas-szewczyk) [inline apps/marketplace/src/graphql/generated/client.ts]: schema.ts adds PRICE_OVERRIDE_REASON_WITHOUT_OVERRIDE, but error-mapping.ts (line 19) lacks the required exhaustive mapping. Storefront and infrastructure type-checks fail.

File: packages/features/src/category/shared/metadata/standard-metadata.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+      description: "Category details",
+    };
+  }
+
+  const rawDescription = category.description;
+  const parsedDescription = editorJSDataToString(rawDescription)?.trim();
+  const ogImageUrl = `${props.storefrontUrl}/categories/${slug}/opengraph-image`;
+
+  return {
+    title: category.seoTitle || category.name,
```
Comment (piotrgrundas) [inline packages/features/src/category/shared/metadata/standard-metadata.ts:44]: Translations?

File: packages/features/src/category/shared/metadata/standard-metadata.ts [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+  if (!category) {
+    return {
+      title: "Category",
+      description: "Category details",
+    };
+  }
+
+  const rawDescription = category.description;
+  const parsedDescription = editorJSDataToString(rawDescription)?.trim();
```
Comment (piotrgrundas) [inline packages/features/src/category/shared/metadata/standard-metadata.ts:40]: Translations?

File: packages/features/src/category/shared/metadata/standard-metadata.ts [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+  const ogImageUrl = `${props.storefrontUrl}/categories/${slug}/opengraph-image`;
+
+  return {
+    title: category.seoTitle || category.name,
+    description:
+      category.seoDescription ??
```
Comment (piotrgrundas) [inline packages/features/src/category/shared/metadata/standard-metadata.ts:46]: Translations?

File: packages/features/src/category/shop-basic-category/standard.tsx [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+    />
+  );
+};
+
+/**
+ * Skeleton component for the standard category view.
+ * This component is used to display a loading state while the category data is being fetched.
+ * @returns A skeleton component for the standard category view.
+ */
+export const StandardCategoryViewSkeleton = () => (
```
Comment (piotrgrundas) [inline packages/features/src/category/shop-basic-category/standard.tsx:115]: Probably identical to the collection skeleton?

File: packages/features/src/collection/shared/metadata/standard-metadata.ts [hunk 6]
```diff
@@ -47,9 +47,10 @@ export async function generateStandardCollectionMetadata(
   return {
     title: collection.seoTitle || collection.name,
     description:
-      collection.seoDescription || parsedDescription?.length
-        ? parsedDescription?.slice(0, 200)
-        : collection.name,
+      collection.seoDescription ??
```
Comment (piotrgrundas) [inline packages/features/src/collection/shared/metadata/standard-metadata.ts:50]: Tranlations here & in tittle

File: packages/infrastructure/src/category/saleor/graphql/fragments/CategoryFragment.graphql [hunk 7]
```diff
@@ -0,0 +1,12 @@
...
+  name
+  slug
+  description
+  seoTitle
+  seoDescription
+  backgroundImage {
+    url
+    alt
+  }
+}
```
Comment (piotrgrundas) [inline packages/infrastructure/src/category/saleor/graphql/fragments/CategoryFragment.graphql:18]: Translations are not pulled at all for seo*, name, description

**Review by piotrgrundas — APPROVED** [review]

**Review by lukas-szewczyk — APPROVED** [review]
